server/internal/decoy/decoy.go

```diff
@@ -385,30 +379,7 @@ func (d *decoy) loadAndDeleteSURBCtx(id uint64) *surbCtx {
 		return nil
 	}
 	delete(d.surbStore, id)
-
-	nCtxList, ok := ctx.etaNode.Value.([]*surbCtx)
-	if !ok {
-		nCtx := ctx.etaNode.Value.(*surbCtx)
-		nCtxList = []*surbCtx{nCtx}
-	}
-	if l := len(nCtxList); l > 1 {
-		// There is more than 1 SURB with this ETA, remove the context from
-		// the list, and leave the node in the tree.
-		for i, v := range nCtxList {
-			if v == ctx {
-				copy(nCtxList[i:], nCtxList[i+1:])
-				nCtxList[l-1] = nil
-				ctx.etaNode.Value = nCtxList[l-1]
-				ctx.etaNode = nil
-				return ctx
-			}
-		}
-		panic("BUG: SURB missing from node ETA list")
-	}
-
-	d.surbETAs.Remove(ctx.etaNode)
-	ctx.etaNode = nil
-
+	d.surbETAs.RemovePriority(ctx.priority)
```

Comment:
check return value of RemovePriority != nil ( see panic("BUG.."))

Reply:
check the return value for what!?

Reply:
this cannot crash. try it.